go/chat/search/session.go

```diff
@@ -105,28 +104,18 @@ func (s *searchSession) percentIndexedLocked() int {
 
 // searchConv finds all messages that match the given set of tokens and opts,
 // results are ordered desc by msg id.
-func (s *searchSession) searchConv(ctx context.Context, convID chat1.ConversationID,
-	convIdx *chat1.ConversationIndex) (msgIDs []chat1.MessageID, err error) {
-	defer s.indexer.Trace(ctx, func() error { return err }, fmt.Sprintf("searchConv convID: %v", convID.String()))()
-	if convIdx == nil {
-		return nil, nil
-	}
-
+func (s *searchSession) searchConv(ctx context.Context, convID chat1.ConversationID) (msgIDs []chat1.MessageID, err error) {
+	defer s.indexer.Trace(ctx, func() error { return err }, fmt.Sprintf("searchConv convID: %s", convID))()
 	var allMsgIDs mapset.Set
 	for token := range s.tokens {
 		matchedIDs := mapset.NewThreadUnsafeSet()
-
-		// first gather the messages that directly match the token
-		for msgID := range convIdx.Index[token] {
-			matchedIDs.Add(msgID)
+		idMap, err := s.indexer.store.GetHits(ctx, s.uid, convID, token)
```

Comment:
could just have this return a `mapset` directly

Reply:
I guess, seemed like just something this guy was using though.

Reply:
don't have to...